[Judy1x4] iP

Judy1x4 opened the pull request:
### Jude The Chatbot Task Assistant

Jude allows you to:
* add tasks in different formats
* delete tasks
* mark tasks as done

The task types include:
1. To-do
2. Deadline 
3. Event

_You can download if you are one of these:_
- [x] aiming for simple task management 🐒
- [x] intuitive Ui of chat environment
- [x] prefer typing over clicking  

>[!NOTE]
>**This product is under the course CS2103T of NUS**. Here's the website you can have a look: [CS2103T Teams](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html/)


If you are a Java programmer, you can use it to practice `Java` too. Here's the main method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Judy1x4 committed:
edit .gitignore

Updated files to be not revision controlled.

Judy1x4 committed:
Rename chatbot.

Replaced all traces of Duke to Jude from the source code.

Judy1x4 committed:
Implement an initial skeletal version

It simply greets the user and exits.

Judy1x4 committed:
Improve the skeletal version

it echos commands entered by the user, and exits when the user types the command bye.

Judy1x4 committed:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

Assume there will be no more than 100 tasks.

Judy1x4 committed:
Correct Java layout and add header comments

Judy1x4 committed:
Task class: Add Task class

Judy1x4 committed:
Task class: Add Javadoc

Judy1x4 committed:
Jude class: Add feature to mark tasks as done or not done

Let's,
* mark a task by an index
* unmark a task by an index

Judy1x4 committed:
Jude class: Add Javadoc of class

Judy1x4 committed:
Create subclasses of Task

There are different tasks which can be grouped into new class.

This enables more specific task management.

Create subclasses of Task with different fields.

Let's
* Deadline class: contains information about deadline
* Event class: contains from and to date/time
* Todo: a simple task without deadline or duration

Using inheritance is preferable due to is-a relationship.

Judy1x4 committed:
Jude class: Add parser object to replace slicing inputs

Judy1x4 committed:
Add Parser class: handles slicing inputs

Judy1x4 committed:
Change reading todo input to to-do input: distinguish with todo comment

Judy1x4 committed:
Use i/o redirection technique to semi-automate the regression testing.

Judy1x4 committed:
JudeException class: handle invalid input

Judy1x4 committed:
Replace code with JudeException where necessary

Judy1x4 committed:
Add invalid test cases

Judy1x4 committed:
Jude: Correct to show full task details when task added

Judy1x4 committed:
Replace Task array to Task Arraylist. Add remove command.

Update test cases and Parser class.

Judy1x4 committed:
Add save feature

Judy1x4 committed:
Merge branch 'branch-Level-7'

Judy1x4 committed:
Work in progress: Replace String date or time in Deadline and Event

Judy1x4 committed:
Complete replacement of String to Time or Date object

Judy1x4 committed:
Work in progress: Combine Date and Time object to DateTime object

Judy1x4 committed:
Complete in Replacing Date and Time object to DateTime object

There are Time and Date objects in Deadline and Event class.

Unifying all those fields improves code quality.

Let's:
* extract those complicated fields into unified DateTime object
* Deadline and Event constructor parameters are simplified
* loadFile split logic is simpler.

Judy1x4 committed:
Ignore jude.txt save file in data directory

Judy1x4 committed:
Merge branch 'branch-Level-8'

Judy1x4 committed:
Add Storage, TaskList, Ui classes and replace main method appropriately

Judy1x4 committed:
Add Command classes

Judy1x4 committed:
Add more methods

Judy1x4 committed:
Debug
Let's
* fix delete command to properly delete the index
* fix invalid index input
* fix typos in command classes
* initialize fields in Jude class

Judy1x4 committed:
Debug index and typo

Let's
* fix accessing deleted command from list, causing an error
* fix index to be zero-indexed in Delete constructor
* fix minor typo

Judy1x4 committed:
Merge branch 'A-MoreOOP'

Judy1x4 committed:
Update jude package to all class and command, task packages

Judy1x4 committed:
Merge branch 'add-gradle-support'

Judy1x4 committed:
Override toString in command classes

Judy1x4 committed:
Fix unhandled exception and typo

Judy1x4 committed:
Add Test classes and write test cases in some classes

Let's
* Add test methods in Parser class and TaskList class

Judy1x4 committed:
Commit build.gradle

Judy1x4 committed:
Commit version control ignored file

Judy1x4 committed:
Add Test classes

Judy1x4 committed:
Merge branch 'A-JUnit'

Judy1x4 committed:
commiting build.gradle

Judy1x4 committed:
Delete files to be untracked by version control

Judy1x4 committed:
Updated .gitignore

Judy1x4 committed:
Write JavaDoc comments for all non-trivial methods and classes

Judy1x4 committed:
Adjust blank line separation and unused field.

Judy1x4 committed:
Replace if-else to switch-cases and fix typos

Judy1x4 committed:
Add find command.

Judy1x4 committed:
Merge branch 'branch-A-JavaDoc'

Judy1x4 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/jude/Parser.java
#	src/main/java/jude/Ui.java

Judy1x4 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/jude/Parser.java

Judy1x4 committed:
Change formatting in Parser class

Judy1x4 committed:
Fix import statement in Parser class

OttoTan commented on src/main/java/jude/command/MarkCommand.java:
```diff
@@ -0,0 +1,36 @@
+    public MarkCommand(int index) {
+        this.index = index;
+    }
+    @Override
```
@Override annotation is on the wrong line

OttoTan commented on src/main/java/jude/command/Command.java:
```diff
@@ -0,0 +1,23 @@
+ */
+public abstract class Command {
+
+    private boolean isExit;
```
The spacing between variables and functions seems wrong here

OttoTan commented on src/main/java/jude/command/FindCommand.java:
```diff
@@ -0,0 +1,18 @@
+        this.keyword = keyword;
+    }
+
+    @Override
```
Perhaps could have a JavaDoc for this function here?

OttoTan reviewed (commented):
It LGTM overall, code is clean and easy to understand 👍

jevan13 commented on src/main/java/jude/Parser.java:
```diff
@@ -0,0 +1,87 @@
+ * This class has the method parse to identify the command based on the input of the user.
+ * If the input is not valid, it throws JudeException.
+ */
+public class Parser {
```
LGTM!

jevan13 commented on src/main/java/jude/Storage.java:
```diff
@@ -0,0 +1,100 @@
+ * It creates a new save file if the save file does not exist.
+ * It saves the file by writing the task datas in the save file, so that it can be loaded the next time.
+ */
+public class Storage {
```
LGTM!

jevan13 commented on src/main/java/jude/TaskList.java:
```diff
@@ -0,0 +1,89 @@
+ * It stores tasks in a list and allows efficient management of different task types, ensuring the users
+ * to keep track of their pending tasks easily.
+ */
+public class TaskList {
```
LGTM!

jevan13 commented on src/main/java/jude/Ui.java:
```diff
@@ -0,0 +1,53 @@
+ * Shows an error if there is any.
+ * Reads input from the user.
+ */
+public class Ui {
```
LGTM!

jevan13 commented on src/main/java/jude/Jude.java:
```diff
@@ -0,0 +1,58 @@
+    public void run() {
+
+        ui.startChat();
+        boolean isExit = false;
```
Is this necessary? You can just terminate the program with exit().

jevan13 commented on src/main/java/jude/task/Event.java:
```diff
@@ -0,0 +1,69 @@
+    private LocalDateTime toDateTime;
+
+    /**
+     * Creates the Deadline object.
```
Should this be Event object instead of Deadline object?

jevan13 commented on src/main/java/jude/task/Event.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+    /**
+     * Creates the Deadline object.
```
Should this be Event object instead of Deadline object?

jevan13 reviewed (commented):
Looks good! Nothing much to change :)

Judy1x4 committed:
Fix index error not matching

Judy1x4 committed:
Replace redundant isExit boolean variable and fix typo in Event header

Fixes according to the pull request comments.

Judy1x4 committed:
Add some of worked Javafx GUI

Judy1x4 committed:
Command class: Store task message

Let's
* set the field message.
* get the field message.
* to be passed as an argument for Jude:getResponse.

Judy1x4 committed:
Add profile images

Judy1x4 committed:
Add label according to the Command type

Let's
* get DialogBox with different color for different commandType.

Judy1x4 committed:
Fix typos

Judy1x4 committed:
Make parse a static method

Judy1x4 committed:
Merge branch 'Level-10'

Judy1x4 committed:
Add Checkstyle and correct the code upto the coding standard.

Judy1x4 committed:
Merge branch 'branch-A-CheckStyle'

Judy1x4 committed:
Create JAR file

Judy1x4 committed:
Remove Ui class and related file containing the Ui variable

Judy1x4 committed:
Remove Ui further from the main class

Judy1x4 committed:
Add package ui and add DialogBox and MainWindow classes

Judy1x4 committed:
Fix program not exiting properly

Judy1x4 committed:
Fix style issues

Judy1x4 committed:
Fix small style issue

Judy1x4 committed:
Merge branch 'branch-Generate-New-File'

Judy1x4 committed:
Add assert in Main class that jude cannot be null

There is an assumption that the default file path is correct. In
some cases where the parent directory is not present, the program
will output IOException when loading the save file.

Asserting the path existence assures the correct code behaviour.

As a step towards such correction, let's assert the jude instance
cannot be null.

Judy1x4 committed:
Refactor methods to adhere to SLAP

There are several methods that has a mix of low-level and
high-level implementation at a same scope. In some cases there
are many lines of low-level codes only.

Refactoring all those low-level code or abstracting with comments
improves the code quality.

As a step towards such refactoring, let's:
* extract those low-level code blocks into seperate methods
in their respective classes.
* add high-level comment to pack in two abstraction levels.
Doing so will make the subsequent coding quality improvement
easier.

Judy1x4 committed:
Import binary operator in TaskList class

Judy1x4 committed:
Replace for loop in Tasklist class with stream

Judy1x4 committed:
Merge pull request #2 from Judy1x4/branch-A-Assertions

Add assert in Main class that jude cannot be null

Judy1x4 committed:
Merge branch 'master' into branch-A-Streams

Judy1x4 committed:
Merge branch 'master' into branch-A-CodeQuality

Judy1x4 committed:
Merge pull request #3 from Judy1x4/branch-A-CodeQuality

Branch a code quality

Judy1x4 committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/jude/TaskList.java

Judy1x4 committed:
Merge pull request #4 from Judy1x4/branch-A-Streams

Replace for loop in Tasklist class with stream

Judy1x4 committed:
Add Sort Command class

Judy1x4 committed:
Merge branch 'branch-BCD-Extension'

Judy1x4 committed:
Correct JavaDoc header to a correct format

Judy1x4 committed:
Debug Junit

Judy1x4 committed:
Fix minor typo

Judy1x4 committed:
Add Ui image

Judy1x4 committed:
Merge branch 'branch-A-UserGuide'

Judy1x4 committed:
Update README.md

Judy1x4 committed:
Update release page.

Judy1x4 committed:
Fix parent directory of data file to be created if not exist

Judy1x4 committed:
Correct code style

Judy1x4 committed:
Correct some style error

Judy1x4 committed:
Fix some more style error

Judy1x4 committed:
Fix style error to not have line separation below class declaration
